app/assets/javascripts/models/data/preferences.js.coffee

```diff
@@ -24,11 +25,16 @@ ns.Preferences = do (ko
         @fromJson(data)
         @isLoaded(true)
       else
-        getJSON '/users/site_preferences', (data, status, xhr) =>
-          console.log "Loaded site preferences, #{JSON.stringify(data)}"
-
-          @fromJson(data)
-          @isLoaded(true)
+        $.ajax '/users/site_preferences',
```

Comment:
The reason for this change, just to proactively explain it, is that there was a race condition with the instantiation of the preferences object.  When preferences is instantiated it is often immediately checked; however, this can get an erroneous result  as the constructor has not completed the getJSON call yet (asynchronous activity).  I had to switch this call to an $.ajax so that I could set async to false for this single call, meaning it will wait to populate itself during instantiation so that when it's immediately called, it has the right goods.